pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -734,36 +794,62 @@ def make_parser(parser: argparse.ArgumentParser):
     return parser
 
 
+def parse_continue_arg(continue_from: Optional[str]) -> Dict[str, Any]:
+    assert (
+        continue_from is None
+        or continue_from == "capture"
+        or re.fullmatch("replay(:[0-9]*)?", continue_from)
+    )
+    result: Dict[str, Any] = {}
+    result["should_prepare_source"] = not continue_from
+    result["should_capture_compile"] = (
+        continue_from is None or continue_from == "capture"
+    )
+    result["should_replay_compile"] = True
+    result["replay_from"] = 1
+    if continue_from and continue_from.startswith("replay:"):
+        result["replay_from"] = int(continue_from.removeprefix("replay:"))
+    return result
+
+
 def main(args):
-    meta_file = Path(args.package[0]).resolve()
     if args.compress_package and not _have_terser():
         raise RuntimeError(
             "Terser is required to compress packages. Try `npm install -g terser` to install terser."
         )
+    step_controls = parse_continue_arg(args.continue_from)
+    # --continue implies --force
+    force = args.force or not not args.continue_from
+
+    meta_file = Path(args.package[0]).resolve()
 
     pkg_root = meta_file.parent
     pkg = parse_package_config(meta_file)
+
+    pkg["source"] = pkg.get("source", {})
+    pkg["build"] = pkg.get("build", {})
+    build_metadata = pkg["build"]
+    build_metadata["cflags"] = build_metadata.get("cflags", "")
+    build_metadata["cxxflags"] = build_metadata.get("cxxflags", "")
+    build_metadata["ldflags"] = build_metadata.get("ldflags", "")
```

Comment:
We should probably improve default parameters handling in config  (and config validation) with something like https://pydantic-docs.helpmanual.io/ but this is OK for now.